src/selectors/entities/preferences.ts

```diff
@@ -89,14 +89,14 @@ export const getVisibleGroupIds = reselect.createSelector(
 export const getTeammateNameDisplaySetting = reselect.createSelector(
     getConfig,
     getMyPreferences,
-    (config, preferences) => {
+    getLicense,
+    (config, preferences, license) => {
+        const useAdminTemmateNameDisplaySetting = (license && license.LockTeammateNameDisplay === 'true') && config.LockTeammateNameDisplay === 'true';
         const key = getPreferenceKey(Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.NAME_NAME_FORMAT);
-        if (preferences[key]) {
+        if (preferences[key] && !useAdminTemmateNameDisplaySetting) {
             return preferences[key].value;
-        } else if (config.TeammateNameDisplay) {
-            return config.TeammateNameDisplay;
         }
-        return General.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME;
```

Comment:
I'm removing this because `config.TeammateNameDisplay` will always have a value associated to it as it gets set in `setDefaults` in server.

https://github.com/mattermost/mattermost-server/blob/master/model/config.go#L2634

Reply:
It's true that this will default to being that value, but I can configure `config.json` to set this value to an empty string and the server won't rewrite it. Should we keep the default to handle this case?

Reply:
Actually, I stand corrected on the substance of this point: the server fails to start if this field is empty. I wrote this comment before testing, then forgot to remove it :D

0/5 on where the default lives, though I think it might still be valuable to be explicit here.

Reply:
Ah my bad! I totally forgot we can do that so I wasn't taking that into account. Addressed!